auth/src/main/java/com/firebase/ui/auth/ui/accountlink/WelcomeBackPasswordPrompt.java

```diff
@@ -122,29 +128,32 @@ private void next(final String email, final String password) {
                 .addOnSuccessListener(new OnSuccessListener<AuthResult>() {
                     @Override
                     public void onSuccess(AuthResult authResult) {
-                        // Get the social AuthCredential from the IDPResponse object, link
-                        // it to the email/password account.
                         AuthCredential authCredential =
                                 AuthCredentialHelper.getAuthCredential(mIdpResponse);
-                        authResult.getUser().linkWithCredential(authCredential);
-                        firebaseAuth.signOut();
```

Comment:
Looks like it was added in [this commit](https://github.com/firebase/FirebaseUI-Android/commit/bf9ef293a9202226b3ab834f8d8a4edb26d7cb45#diff-a21268aefa41be1f56b69905b4679cc3)

The only reason I can think to do this is that known issue about `FirebaseUser` changes not taking effect until signout/signin, but I believe that's only for explicit profile information change requests.

One thing I am seeing now is that the `linkWithCredential` call is done as fire-and-forget, which will almost certainly create some strange race conditions.

Seems like the original author should have been chaining the link-->signout-->sign in operations together.